storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1370,9 +1469,974 @@ public static int toPositive(int number) {
+            LOG.info("IOException Error when trying to kill " + pid + ".");
+        } finally {
+            return retval;
+        }
+    }
+
+    public static int forceKillProcess (long pid) {
+        return sendSignalToProcess(pid, SIGKILL);
+    }
+
+    public static int forceKillProcess (String pid) {
+        return sendSignalToProcess(Long.parseLong(pid), SIGKILL);
+    }
+
+    public static int killProcessWithSigTerm (long pid) {
+        return sendSignalToProcess(pid, SIGTERM);
+    }
+    public static int killProcessWithSigTerm (String pid) {
+        return sendSignalToProcess(Long.parseLong(pid), SIGTERM);
+    }
+
+    /*
+        Adds the user supplied function as a shutdown hook for cleanup.
+        Also adds a function that sleeps for a second and then sends kill -9
+        to process to avoid any zombie process in case cleanup function hangs.
+     */
+    public static void addShutdownHookWithForceKillIn1Sec (Runnable func) {
+        Runnable sleepKill = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Time.sleepSecs(1);
+                    Runtime.getRuntime().halt(20);
+                } catch (Exception e) {
+                    LOG.warn("Exception in the ShutDownHook: " + e);
+                }
+            }
+        };
+        Runtime.getRuntime().addShutdownHook(new Thread(func));
+        Runtime.getRuntime().addShutdownHook(new Thread(sleepKill));
+    }
+
+    /**
+     * Returns the combined string, escaped for posix shell.
+     * @param command the list of strings to be combined
+     * @return the resulting command string
+     */
+    public static String shellCmd (List<String> command) {
+        List<String> changedCommands = new LinkedList<>();
+        for (String str: command) {
+            if (str == null) {
+                continue;
+            }
+            changedCommands.add("'" + str.replaceAll("'", "'\"'\"'") + "'");
+        }
+        return StringUtils.join(changedCommands, " ");
+    }
+
+    public static String scriptFilePath (String dir) {
+        return dir + FILE_PATH_SEPARATOR + "storm-worker-script.sh";
+    }
+
+    public static String containerFilePath (String dir) {
+        return dir + FILE_PATH_SEPARATOR + "launch_container.sh";
+    }
+
+    public static void throwRuntime (Object... strings) {
+        String combinedErrorMessage = "";
+        for (Object oneMessage: strings) {
+            combinedErrorMessage = combinedErrorMessage + oneMessage.toString();
+        }
+        throw new RuntimeException(combinedErrorMessage);
+    }
+
+    public static Object nullToZero (Object v) {
+        return (v!=null? v : 0);
+    }
+
+    public static Object containerGet (Container container) {
+        return container.object;
+    }
+
+    public static Container containerSet (Container container, Object obj) {
+        container.object = obj;
+        return container;
+    }
```

Comment:
Why the crap do we even have a Container class?  I guess it is OK, but lets move these to be set and get methods on Container itself, and then use the proper type hints.